sdk/storage/azure-storage-blob/azure/storage/blob/_shared/uploads.py

```diff
@@ -477,6 +477,13 @@ def read(self, size=None):
                             raise IOError("Stream failed to seek to the desired location.")
                         buffer_from_stream = self._wrapped_stream.read(current_max_buffer_size)
                 else:
+                    absolute_position = self._stream_begin_index + self._position
+                    # It's possible that there's connection problem during data transfer,
+                    # so when we retry we don't want to read from current position of wrapped stream,
+                    # instead we should seek to where we want to read from.
+                    if self._wrapped_stream.tell() != absolute_position:
+                        self._wrapped_stream.seek(absolute_position, SEEK_SET)
```

Comment:
note to self: we could also always seek but this is bad because it wouldn't be as efficient (as Emily explained offline)